pkg/registry/generic/etcd/etcd.go

```diff
@@ -441,6 +441,9 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	// delete immediately, or no graceful deletion supported
 	out := e.NewFunc()
 	if err := e.Storage.Delete(ctx, key, out); err != nil {
+		if storage.IsNotFound(err) {
+			return nil, nil
```

Comment:
also, we can't return a nil result here... shouldn't we return the same thing we do in the `alreadyDeleting` case above, e.g. `return e.finalizeDelete(obj, true)`?